### PR TITLE
improve case handling for ssh auth

### PR DESCRIPTION
**Pull request by p4ndepravity**
also provide custom auth method capabilities on yml

### CODE REVIEW

File: lib/kitchen/transport/ssh.rb [hunk 1]
```diff
@@ -415,23 +415,28 @@ def connection_options(data)
...
         end
 
-        opts[:password] = data[:password]           if data.key?(:password)
         opts[:forward_agent] = data[:forward_agent] if data.key?(:forward_agent)
         opts[:verbose] = data[:verbose].to_sym      if data.key?(:verbose)
 
         # disable host key verification. The hash key to use
         # depends on the version of net-ssh in use.
-        opts[verify_host_key_option] = false
+        opts[:verify_host_key_option] = false
```
Comment (coderanger) [inline lib/kitchen/transport/ssh.rb]: That was a method, not a symbol.

**Comment by coderanger** [comment]
Can you be more specific about your goal here?

**Comment by p4ndepravity** [comment]
I have been unable to successfully authenticate using ssl for the ssh connection so I needed to provide a means for using password authentication. also there was a situation where an ssh key was provided (by kitchen by default) as well as a password and the if data[:ssh_key] && !data[:password] block wasn't being executed.

**Comment by cheeseplus** [comment]
As it stands this is broken and I'm still unclear what problem this is solving for so closing this one but happy to review another PR with a clearer description of the problem statement and how this solves it.

**Pull request closed without merging**